contracts/contracts/validator-registry/middleware/MevCommitMiddleware.sol

```diff
@@ -224,44 +224,41 @@ contract MevCommitMiddleware is IMevCommitMiddleware, MevCommitMiddlewareStorage
+    /// @param captureTimestamps block.timestamps of the latest finalized block that the blsPubkey was queried as "opted-in" by the oracle.
+    function slashValidators(bytes[] calldata blsPubkeys, uint256[] calldata captureTimestamps) external onlySlashOracle {
         uint256 len = blsPubkeys.length;
-        require(len == infractionTimestamps.length, InvalidArrayLengths(len, infractionTimestamps.length));
+        require(len == captureTimestamps.length, InvalidArrayLengths(len, captureTimestamps.length));
 
         address[] memory swappedOperators = new address[](len);
         address[] memory swappedVaults = new address[](len);
         uint256[] memory newPositions = new uint256[](len);
 
         for (uint256 i = 0; i < len; ++i) {
             bytes calldata pubkey = blsPubkeys[i];
-            // These and other checks in _slashValidator will succeed if current tx executes within
-            // slashPeriodSeconds of validator being marked as "not opted-in",
-            // OR relevant validator/vault/operator has not fully deregistered yet.
+            // These and other checks in _slashValidator are guaranteed to succeed if current tx executes within
+            // slashPeriodSeconds of a validator's captureTimestamp (defined in README.md).
             ValidatorRecord storage valRecord = validatorRecords[pubkey];
             require(valRecord.exists, MissingValidatorRecord(pubkey));
-            // Store slash record if it doesn't already exist. To ensure desirable ordering, _getNumSlashableVals should 
-            // intentionally be computed once for the slash record, as collateral that is slashed later in this function
-            // affects the metric.
+
             SlashRecord storage slashRecord = slashRecords[valRecord.vault][valRecord.operator][block.number];
             if (!slashRecord.exists) {
-                uint256 numSlashableRegistered = _getNumSlashableRegisteredVals(valRecord.vault, valRecord.operator);
-                require(numSlashableRegistered != 0, ValidatorsNotSlashable(valRecord.vault, valRecord.operator, len, numSlashableRegistered));
+                uint256 numRegistered = _vaultAndOperatorToValset[valRecord.vault][valRecord.operator].length();
+                require(numRegistered != 0, NoRegisteredValidators(valRecord.vault, valRecord.operator));
                 slashRecords[valRecord.vault][valRecord.operator][block.number] = SlashRecord({
                     exists: true,
                     numSlashed: 0,
-                    numInitSlashableRegistered: numSlashableRegistered
+                    numRegistered: numRegistered
                 });
             }
-            // Swap about to be slashed pubkey with last slashable pubkey in valset.
-            uint256 newPosition = slashRecord.numInitSlashableRegistered - slashRecord.numSlashed; // 1-indexed
+            // Swap about to be slashed pubkey with last pubkey in registered valset.
```

Comment:
OK. But wouldn't the last registered be at position `slashRecord.numRegistered`? But since we might slash multiple ones here, we don't want to swap the second one to the same position, that's why it's `slashRecord.numRegistered - slashRecord.numSlashed`. So technically the second one who gets slashed is not swapped to the last registered etc. If my understanding is correct, maybe a good formulation would be something like "Swap about to be slashed pubkey to the end of registered valset, before previously slashed ones."